libosrmc/osrmc.cc

```diff
@@ -112,6 +113,32 @@ void osrmc_params_add_coordinate_with(osrmc_params_t params, float longitude, fl
   osrmc_error_from_exception(e, error);
 }
 
+/* Route service */
+osrmc_route_annotations_t osrmc_route_annotations_construct(osrmc_error_t* error) try {
+  auto* out = new osrm::RouteParameters::AnnotationsType{osrm::RouteParameters::AnnotationsType::Duration};
+  return reinterpret_cast<osrmc_route_annotations_t>(out);
+} catch (const std::exception& e) {
+  osrmc_error_from_exception(e, error);
+  return nullptr;
+}
+
+void osrmc_route_annotations_destruct(osrmc_route_annotations_t annotations) {
+  delete reinterpret_cast<osrm::RouteParameters::AnnotationsType*>(annotations);
+}
+
+void osrmc_route_annotations_enable_distance(osrmc_route_annotations_t annotations, bool enable, osrmc_error_t* error) try {
+  using AnnotationsType = osrm::RouteParameters::AnnotationsType;
+  auto* annotations_typed = reinterpret_cast<AnnotationsType*>(annotations);
+
+  if (enable) {
+    *annotations_typed |= AnnotationsType::Distance;
+  } else {
+    *annotations_typed = static_cast<AnnotationsType>(static_cast<int>(*annotations_typed) ^ static_cast<int>(AnnotationsType::Distance));
+  }
+} catch (const std::exception& e) {
+  osrmc_error_from_exception(e, error);
+}
```

Comment:
Why are these annotation features needed for the trip service?

If you want them in, can you split it at least into two commits please.